std/format.d

```diff
@@ -2628,55 +2632,118 @@ if (is(FloatingPointTypeOf!T) && !is(T == enum) && !hasToString!(T, Char))
     const nan = isNaN(tval);
     const inf = isInfinity(tval);
 
-    if (nan || inf)
+    static if (is(T == float) || is(T == double) || (is(T == real) && T.mant_dig == double.mant_dig))
     {
```

Comment:
I first thought to do this, but it's a little bit difficult, because of a mix of `static if` and `if`. I would probably need to put everything in a separate function. But as this is only temporarily needed (until all specifiers are implemented) I think it's better to keep it simple here.

Reply:
> keep it simple here.

There is a difference between simple to write and simple to read. In this case I am trying to figure out what the difference between the two branches are

For this case 
```
size_t len;
char[] buf;
if (fs.spec=='a' || fs.spec=='A')
{
    static if(...)
    {
        buf = printFloat(val, fs, FloatingPointControl.rounding).dup;
        len = buf.length;
    }
    else
        goto notFloating;
}
else
{
    notFloating:
    ...
}
```
will work.

Reply:
Nice trick with that goto. I tried it and it works.